### PR TITLE
Fix importing. Use .xcdatamodeld for coredata

### PR DESCRIPTION
**Pull request by ZachGawlik**
@leemcdole @baylee 
Made a lot of small changes so that it would work well for the web interface. 

* Edited __init__ and __main__ so that `run_signals` function could be imported
* Added `check_xcode` flag variable so that it wouldn't error when running the process from a server
* Fixed bug when `helpers.py` doesn't find the parent file
* Switched `core_data` input to take the `.xcdatamodeld` folder so that the user doesn't have to go through multiple "show package contents" clicks to get the path. If there are multiple coredata versions, it uses the current version as set in Xcode.

### CODE REVIEW

File: signals/generators/ios/core_data.py [hunk 1]
```diff
@@ -199,12 +200,33 @@ def get_model(xml):
...
-    model = get_model(xml)
+# parses the hidden .xccurrentversion for the name of the current version
+# this information is stored inside its first and only <string> xml tag
+def get_current_version(xcdatamodeld_path):
+    xccurrentversion_path = xcdatamodeld_path + '/.xccurrentversion'
+
+    # .xccurrentversion might not exist if there is only one version
+    if os.path.exists(xccurrentversion_path):
+        xml_dom = minidom.parse(xccurrentversion_path)
+        return xml_dom.getElementsByTagName('string')[0].childNodes[0].data
```
Comment (rmutter) [inline signals/generators/ios/core_data.py]: should there be any extra error checking around making sure the xml tag exists or are we not concerned?

File: signals/generators/ios/core_data.py [hunk 2]
```diff
@@ -199,12 +200,33 @@ def get_model(xml):
...
+    else:
+        return [f for f in os.listdir(xcdatamodeld_path) if f.endswith('.xcdatamodel')][0]
```
Comment (rmutter) [inline signals/generators/ios/core_data.py]: does making `f` named `file` makes this statement past our PEP line length?

File: signals/generators/ios/core_data.py [hunk 3]
```diff
@@ -199,12 +200,33 @@ def get_model(xml):
...
+
+
+def get_core_data_from_folder(xcdatamodeld_path):
+    current_version_name = get_current_version(xcdatamodeld_path)
+    core_data_path = xcdatamodeld_path + '/' + current_version_name + '/contents'
```
Comment (rmutter) [inline signals/generators/ios/core_data.py]: might be better to use the python string formatting instead of appending together with `+`

**Comment by rmutter** [comment]
@ZachGawlik couple of small comments above and what we just chatted about - running both `signals` after pip installing it and `python -m signals` locally when developing

**Comment by ZachGawlik** [comment]
@rmutter Made PR changes, and updated readme command.

**Comment by rmutter** [comment]
@ZachGawlik looks good :+1:

**Pull request merged**